content/using_ec2_spot_instances_with_eks/spotmanagednodegroups/workers_eksctl.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Adding Spot Workers with eksctl"
+date: 2018-08-07T11:05:19-07:00
+weight: 30
+draft: false
+---
+
+In this section we will deploy the instance types we selected and request nodegroups that adhere to Spot diversification best practices. For that we will use **[eksctl create nodegroup](https://eksctl.io/usage/managing-nodegroups/)** to generate the ClusterConfig file and save the output. The ClusterConfig file will be edited with additional configuration including filtering the instance types, and use the edited eksctl configuration files to add the new nodes to the cluster.
+
+Let's first create the configuration file:
+
+```bash
+eksctl create nodegroup \
+    --cluster=eksworkshop-eksctl \
+    --region=$AWS_REGION \
+    --dry-run \
+    --managed \
+    --spot \
+    --name=dev-4vcpu-16gb-spot \
+    --instance-selector-vcpus=4 \
+    --instance-selector-memory=16 \
+    --instance-selector-cpu-architecture=x86_64 \
+    --instance-selector-gpus=0 \
+    > ~/environment/spot_nodegroup_4vcpu_16gb.yml
+
+eksctl create nodegroup \
+    --cluster=eksworkshop-eksctl \
+    --region=$AWS_REGION \
+    --dry-run \
+    --managed \
+    --spot \
+    --name=dev-8vcpu-32gb-spot \
+    --instance-selector-vcpus=8 \
+    --instance-selector-memory=32 \
+    --instance-selector-cpu-architecture=x86_64 \
+    --instance-selector-gpus=0 \
+    > ~/environment/spot_nodegroup_8vcpu_32gb.yml
+```
+
+This will create 2 files, `spot_nodegroups_4vcpu_16gb.yml` and `spot_nodegroups_8vcpu_32gb.yml`, that we will use to instruct eksctl to create two nodegroups, both with a diversified configuration.
+
+Let's edit the 2 configuration files before using it to create the node groups:
+
+1. Change the **maxSize** to 5 and **minSize** to 1. Leave **desiredCapacity** at 2. 
+
+1. Delete the *d3en*, *h1*, *m5zn* and *g4dn* instances from the **instanceTypes**. Leave in only the Instances from M and T families (except m5zn.large and m5zn.xlarge).
+
+1. In the **labels:** section, add the following labels in a new line: **intent: apps**
+
+    We will use this label **intent** to force a hard partition of the cluster for our applications. During this workshop we will deploy control applications on nodes that have been labeled with **intent: control-apps** while our applications get deployed to nodes labeled with **intent: apps**.
+
+1. In the  **tags:** section, add the following tags in a new line: **k8s.io/cluster-autoscaler/node-template/label/intent: apps**
+
+1. Remove the section **instanceSelector:**.
+
+If you are still struggling with the implementation, the solution files is available here
+
+Note: The region in the solution files is set to us-west-2. Remember to change the region to *your region* if you are not using us-west-2.
```

Comment:
we can't just have the $AWS_REGION parameter in there?

Reply:
Changed to use $AWS_REGION parameter and added a note to manually change to the appropriate region.